src/runtime/hexagon_dma_device_interface.cpp

```diff
@@ -0,0 +1,441 @@
+    if(halide_hexagon_dmart_get_frame_index(user_context, (uintptr_t)buf->host, &frame_idx)) {
+        error(NULL) << "Function failed to get Frame Index  \n";
+        return E_ERR;
+    }
+    /*Get the Dma Context*/
+    if(halide_hexagon_dmart_get_context (user_context, (void**) &dma_context)) {
+        error(NULL) << "Function failed to get DMA Context \n";
+        return E_ERR;
+    }
+    /*Get the Chroma and Luma Type*/
+    chroma_type = dma_context->presource_frames[frame_idx].chroma_type;
+    luma_type = dma_context->presource_frames[frame_idx].luma_type;
+
+    t_eDmaFmt ae_fmt_id[2] = {luma_type,chroma_type};
+    int roi_width = 0;
+    int roi_height = 0;
+    bool is_ubwc = 0;
+    bool padding = 0;
+    int frame_width = 0;
+    int frame_height = 0;
+    int frame_stride = 0;
+    int luma_stride = 0;
+    int chroma_stride = 0;
+    void* dma_handle = NULL;
+    int ncomponents;
+    int nfolds;
+    int plane;
+
+    /* Get the necessary Parameters from DMA Context*/
+    /*Get ROI and Padding Details */
+    roi_width = dma_context->presource_frames[frame_idx].fold_width;
+    roi_height = dma_context->presource_frames[frame_idx].fold_height;
+    padding = dma_context->presource_frames[frame_idx].padding;
+    is_ubwc = dma_context->presource_frames[frame_idx].is_ubwc;
+    /*Get the Frame Details*/
+    frame_width = dma_context->presource_frames[frame_idx].frame_width;
+    frame_height = dma_context->presource_frames[frame_idx].frame_height;
+    frame_stride = dma_context->presource_frames[frame_idx].frame_stride;
+    plane = dma_context->presource_frames[frame_idx].plane;
+    ncomponents = ((plane==LUMA_COMPONENT)|| (plane==CHROMA_COMPONENT))?1:2;
+    nfolds = dma_context->presource_frames[frame_idx].num_folds;
+
+    t_dma_pix_align_info pst_roi_size;
+    pst_roi_size.u16W = roi_width;
+    pst_roi_size.u16H = roi_height;
+    // Note: Both the source and destination are checked for UBWC mode
+    // as buffers are shared between the source frame and destination frame.
+    luma_stride = dma_get_stride(luma_type, is_ubwc, pst_roi_size);
+    chroma_stride = dma_get_stride(chroma_type, is_ubwc, pst_roi_size);
+
+    //#######################################################
+    //Allocate DMA if required
+    //######################################################
+    bool dma_allocate;
+    if(halide_hexagon_dmart_allocate_dma(user_context,(uintptr_t)buf->host, &dma_allocate)) {
+        error(NULL) << "Function failed to check if DMA Alloc is needed \n";
+        return E_ERR;
+    }
+
+    if(dma_allocate) {
+        /* No Free DMA Engine Available, Allocate one */
+        dma_handle = dma_allocate_dma_engine();
+        if (dma_handle == 0) {
+            error(NULL) << "halide_hexagon_dma_device_malloc:Failed to allocate the read DMA engine.\n";
+            return E_ERR;
+        }
+        if(halide_hexagon_dmart_set_dma_handle(user_context, dma_handle,(uintptr_t)buf->host)) {
```

Comment:
There should be a space between `if` and `(`. The code in this PR is inconsistent, sometimes does this, sometimes not.

The same goes for spaces between arguments, this line has a space after the `user_context`, but not after `dma_handle`. There should be a space (like `user_context`).